app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskScreen.kt

```diff
@@ -84,9 +85,10 @@ fun AddEditTaskScreen(
         )
 
         // Check if the task is saved and call onTaskUpdate event
-        LaunchedEffect(uiState.isTaskSaved) {
-            if (uiState.isTaskSaved) {
-                onTaskUpdate()
+        if (uiState.isTaskSaved) {
```

Comment:
This is using recomposition as an observer of state change events to apply other changes to app state. Avoid this.  Observe the state changes inside the `LaunchedEffect` instead so that it doesn't take one recomposition pass to apply `currentOnTaskUpdate` and then another to apply changes caused by that call.